app/react/Settings/components/ToggleChildren.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useState } from 'react';
+import { ToggleButton } from 'app/UI';
+
+export interface ToggleChildrenProps {
+  toggled: boolean;
+  children?: JSX.Element[] | JSX.Element | string;
+  showChildren?: boolean;
+  onToggleOff?: () => void;
+  onToggleOn?: () => void;
+}
+
+export const ToggleChildren = ({
+  children,
+  toggled = false,
+  onToggleOff = () => {},
+  onToggleOn = () => {},
+}: ToggleChildrenProps) => {
+  const [isToggled, setToggled] = useState(toggled);
+
+  const handleClick = () => {
+    if (isToggled) {
+      onToggleOff();
+    } else {
+      onToggleOn();
+    }
+    setToggled(!isToggled);
+  };
+
+  return (
+    <>
+      <div className="toggle-children-button">
+        <ToggleButton onClick={handleClick} checked={isToggled} />
+      </div>
+      <div className="toggle-children-children" style={{ display: isToggled ? 'block' : 'none' }}>
```

Comment:
@RafaPolit @Zasa-san this is still rendering the elements themselves, not a big issue but i think it should be preferabe to not render instead of display: none ?

Reply:
No.  Actually, that removed the reference on the form, so they need to be rendered without display or the bind between the input and the form hook is lost.  What you mention was our first approach and it didn't work, unfortunately. :(